components/kyma-environment-broker/internal/provisioner/fake_client.go

```diff
@@ -82,6 +82,19 @@ func (c *FakeClient) FindOperationByRuntimeIDAndType(runtimeID string, operation
 	return schema.OperationStatus{}
 }
 
+func (c *FakeClient) FindLastOperationByRuntimeIDAndType(runtimeID string, operationType schema.OperationType) schema.OperationStatus {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	var opStatus schema.OperationStatus
+	for _, status := range c.operations {
+		if *status.RuntimeID == runtimeID && status.Operation == operationType {
+			opStatus = status
```

Comment:
nit: if iterated from the end, you could break the cycle here and save a few CPU instructions :)